src/lib.rs

```diff
@@ -381,6 +382,39 @@ async fn delete_habit(
     Ok(HttpResponse::Ok().finish())
 }
 
+#[put("/habit/{habit}")]
+async fn edit_habit(
+    req: HttpRequest,
+    new_name: web::Json<Habit>,
+    path: web::Path<String>,
+    state: web::Data<State>,
+) -> Result<HttpResponse, CustomError> {
+    info!("Edit habit");
+    let habit = path.into_inner();
+    let username = authenticate(req.headers()).map_err(CustomError::AuthError)?;
+
+    let mut stmt_result = state.conn.lock().expect("failed to lock conn");
+    let conn = &mut *stmt_result;
+
+    let result = conn.execute(
+        "UPDATE habit SET name=?1 WHERE username=?2 AND name=?3",
+        (&new_name.name, &username, &habit),
+    );
+    match result {
+        Ok(_) => {
```

Comment:
Let's make sure that the number of affected rows is 1, and return an error if not.

As described in the [docs](https://docs.rs/rusqlite/latest/rusqlite/struct.Connection.html#method.execute), result should contain the number of affected rows.
So we have to do something like:
```
Ok(updated) => {
    if updated == 0 {
            info!("error editing habit, habit not found", e);
            return Err(CustomError::NotFound(anyhow::anyhow!(
                "Habit not found"
            )));
    }
    info!("edited habit");
}
```